realsense2_camera/src/base_realsense_node.cpp

```diff
@@ -886,6 +893,51 @@ IMUInfo BaseRealSenseNode::getImuInfo(const rs2::stream_profile& profile)
     return info;
 }
 
+void BaseRealSenseNode::fillMessageImage(
+    cv::Mat& cv_matrix_image,
+    const stream_index_pair& stream,
+    unsigned int width,
+    unsigned int height,
+    unsigned int bpp,
+    const rclcpp::Time& t,
+    sensor_msgs::msg::Image* img_msg_ptr)
+{
+    // Convert the CV::Mat into a ROS image message (1 copy is done here)
+    cv_bridge::CvImage(std_msgs::msg::Header(), _encoding.at(bpp), cv_matrix_image).toImageMsg(*img_msg_ptr);
+
+    // Convert OpenCV Mat to ROS Image
+    img_msg_ptr->header.frame_id = OPTICAL_FRAME_ID(stream);
+    img_msg_ptr->header.stamp = t;
+    img_msg_ptr->height = height;
+    img_msg_ptr->width = width;
+    img_msg_ptr->is_bigendian = false;
+    img_msg_ptr->step = width * bpp;
+}
+
+cv::Mat& BaseRealSenseNode::getCVMatImage(
+    rs2::frame& frame,
+    std::map<stream_index_pair, cv::Mat>& images,
+    unsigned int width,
+    unsigned int height,
+    unsigned int bpp,
+    const stream_index_pair& stream)
+{
+    auto& image = images[stream];
+    if (image.size() != cv::Size(width, height) || image.depth() != _image_formats[bpp])
```

Comment:
`image.depth()` return the format?

Reply:
Thanks

Reply:
but doesn't bpp stand for bits per pixel? To me it looks confusing that it means an index to array of image formats.